packages/next/src/server/next-server.ts

```diff
@@ -339,6 +340,12 @@ export default class NextNodeServer extends BaseServer {
 
     // ensure options are set when loadConfig isn't called
     setHttpClientAndAgentOptions(this.nextConfig)
+
+    // Intercept fetch
+    if (this.serverOptions.experimentalTestProxy) {
+      const { createTestFetch } = require('./test')
+      global.fetch = createTestFetch(global.fetch)
```

Comment:
Should this be done in `base-server` to ensure it applies for edge-runtime as well?

Reply:
For now, I'd rather keep it strictly to the local-only testing, since proxying can run into all kinds of risks. Thus, basically, only `next dev` and `next start` scenarios.